src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.exception.NoSuchAnnotationException;
+import core.basesyntax.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+    private static final Class MARKER = Dao.class;
+
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException, InstantiationException {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                field.setAccessible(true);
+                if (!(Factory.getBetDao().getClass().isAnnotationPresent(MARKER))
+                        || !(Factory.getUserDao().getClass().isAnnotationPresent(MARKER))) {
+                    throw new NoSuchAnnotationException("There are no such anotation" + MARKER);
```

Comment:
```suggestion
                    throw new NoSuchAnnotationException("There is no such anotation" + MARKER);
```
let's make message more informative and add class that is missing required annotation